docs/core-protocol/public-lock/minting-keys.md

```diff
@@ -0,0 +1,37 @@
+---
+title: Minting Keys
+description: Membership NFTs are called keys and they can be minted from the Lock contract using two different methods.
+sidebar_position: 2
+---
+
+In the Unlock Protocol ecosystem, Membership Non Fungible Tokens are called [Keys](https://unlock-protocol.com/guides/locksmith-lingo/). Keys are minted from the Lock contract.
+
+There are two methods that can be used to mint Keys. Keys can either be **purchased** or **granted**. Keys can be minted one-by-one or in batches (multiple Keys minted at one time) using either method.
+
+### Purchasing Keys
+
+Let's start by noting that this flow uses the function [`purchase`](/core-protocol/smart-contracts-api/PublicLock#purchase) from the lock even if the keys are free. The best way to describe this flow is that it is _self-served_, which means that anyone can call this function to mint an NFT, whether these keys are free or paid.
+
+:::info
+Note: To disable purchases completely, the best approach is to set the maximum number of keys available for sale to be `0`.
+:::
+
+The `purchase` function can be used to purchase multiple keys at once, and the function takes arguments which are all `arrays` in order to accommodate this. Each of these arrays needs to be of the same size.
+
+One of the arguments is an array of recipients, which means that the wallet sending the transaction can be _different_ from the one who will receive the NFT memberships. This can be very useful when application is buying on behalf of users for examples, or when a payment happens "off-chain" (using credit card or other mechanism) and the payment provider then wants to mint the NFT membership to the user.
```

Comment:
```suggestion
One of the arguments is an array of recipients, which means that the wallet sending the transaction can be _different_ from the one who will receive the NFT memberships. This can be very useful when, for example, the application is buying on behalf of users or when a payment happens "off-chain" (using credit card or other mechanism) and the payment provider then wants to mint the NFT membership to the user.
```